Kerberos.NET/Client/Krb5TicketCache.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System.Diagnostics;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Kerberos.NET.Client
+{
+    [DebuggerDisplay("{cache}")]
+    public class Krb5TicketCache : TicketCacheBase
+    {
+        private readonly string filePath;
+        private readonly Krb5CredentialCache cache;
+
+        private readonly object fileSync = new object();
+
+        public Krb5TicketCache(string filePath, ILoggerFactory logger = null)
+            : this(logger)
+        {
+            this.filePath = filePath;
+            this.ReadCache();
+        }
+
+        public Krb5TicketCache(byte[] cache, ILoggerFactory logger = null)
+            : this(logger)
+        {
+            this.ReadCache(cache);
+        }
+
+        protected Krb5TicketCache(ILoggerFactory logger)
+            : base(logger)
+        {
+            this.cache = new Krb5CredentialCache();
+            this.cache.Header[Krb5CredentialCacheTag.KdcClientOffset] = new byte[8];
+        }
+
+        private void ReadCache(byte[] cache)
+        {
+            if (cache == null || cache.Length <= 0)
```

Comment:
```suggestion
            if (cache == null || cache.Length == 0)
```
`Length` can't be negative.

Or use
```c#
if ((cache?.Length ?? 0) == 0)
{
    return;
}
```
Although this is somewhat cryptic to read...

Reply:
Old habit. Length is int, which can be negative, despite never logically being negative.